msal/proguard-rules.pro

```diff
@@ -15,3 +15,36 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
+
+##---------------Begin: proguard configuration for MSAL  --------
+-keep class com.microsoft.** { *; }
```

Comment:
It's not really the idea behind a Proguard configuration file to keep *all* files of the library. You're supposed to only keep classes that are accessed through JNI or reflection: 
https://developer.android.com/studio/build/shrink-code#keep-code

Otherwise, it removes the purpose of shrinking and obfuscation.

Reply:
Sure - unfortunately I don't have the test resources to verify that all components in the `com.microsoft.*` namespace can be shrunk and still behave normally, as we have some dependencies in this namespace that are external to our team. A full test pass takes a few days and a few people

Maybe at some point in the future this can be done in an exact way, but this is today's point-in-time approach